tests/test_init_and_backend.py

```diff
@@ -0,0 +1,70 @@
+from os import error
+from pathlib import Path
+
+import mkosi
+import mkosi.backend as backend
+
+
+class TestMkosiMethods(unittest.TestCase):
+    def test_workspace(self):
+        self.assertEqual(backend.workspace(Path('/home/folder/mkosi/mkosi')), Path('/home/folder/mkosi'))
+        self.assertEqual(backend.workspace(Path('/home/../home/folder/mkosi/mkosi')), Path('/home/../home/folder/mkosi'))
+        self.assertEqual(backend.workspace(Path('/')), Path('/'))
+        self.assertEqual(backend.workspace(Path()), Path())
+
+    def test_fedora_release_cmp(self):
+        self.assertEqual(mkosi.fedora_release_cmp("rawhide", "rawhide"), 0)
+        self.assertEqual(mkosi.fedora_release_cmp("32", "32"), 0)
+        self.assertTrue(mkosi.fedora_release_cmp("33", "32") > 0)
+        self.assertTrue(mkosi.fedora_release_cmp("30", "31") < 0)
+        self.assertTrue(mkosi.fedora_release_cmp("-1", "-2") > 0)
+        self.assertTrue(mkosi.fedora_release_cmp("1", "-2") > 0)
+        with self.assertRaises(ValueError):
+            mkosi.fedora_release_cmp("literal", "rawhide")
+
+    def test_strip_suffixes(self):
+        self.assertEqual(mkosi.strip_suffixes(Path("home/test.zstd")), Path("home/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home/test.xz")), Path("home/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home/test.raw")), Path("home/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home/test.tar")), Path("home/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home/test.cpio")), Path("home/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home/test.qcow2")), Path("home/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home.xz/test.xz")), Path("home.xz/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home.xz/test")), Path("home.xz/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home.xz/test.txt")), Path("home.xz/test.txt"))
+
+    def test_copy_file(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            dir_path = Path(tmp_dir)
+
+            file_1 = Path(dir_path.joinpath("file_1.txt")).touch()
+            file_2 = Path(dir_path.joinpath("file_2.txt")).touch()
+
+            # Copying two empty files
+            mkosi.copy_file(dir_path.joinpath("file_1.txt"), dir_path.joinpath("file_2.txt"))
+            self.assertTrue(filecmp.cmp(dir_path.joinpath("file_1.txt"), dir_path.joinpath("file_2.txt"), shallow=True))
+
+            # Copying content from one file
+            with open(dir_path.joinpath("file_1.txt"), 'w') as file_1:
+                file_1.write("Testing copying content from this file to file_2.")
+            mkosi.copy_file(dir_path.joinpath("file_1.txt"), dir_path.joinpath("file_2.txt"))
+            self.assertTrue(filecmp.cmp(dir_path.joinpath("file_1.txt"), dir_path.joinpath("file_2.txt"), shallow=True))
+
+            # Giving a non existing path/file
+            with self.assertRaises(error):
```

Comment:
I think error here can be replaced with `OSError`.

Reply:
Turns out it is ValueError!

Reply:
Then I'd suggest changing it to ValueError (maybe you forgot to push?)